docs/Compatibility.md

```diff
@@ -4,156 +4,247 @@ The Ruby Datadog Trace library is open source. See the [dd-trace-rb][1] GitHub r
+|--------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| <span id="support-unsupported">Unsupported</span>      | No implementation. [Contact our customer support team for special requests.](https://www.datadoghq.com/support/)                                                             |
+| <span id="support-beta">Beta</span>                    | Initial implementation. May not yet contain all features. Support for new features, bug & security fixes provided on a best-effort basis.                                    |
+| <span id="support-ga">General Availability (GA)</span> | Full implementation of all features. Full support for new features, bug & security fixes.                                                                                    |
+| <span id="support-maintenance">Maintenance</span>      | Full implementation of existing features. Does not receive new features. Support for critical bug & security fixes only.                                                     |
+| <span id="support-legacy">Legacy</span>                | Legacy implementation. May have limited function, but no maintenance provided. [Contact our customer support team for special requests.](https://www.datadoghq.com/support/) |
+| <span id="support-eol">End-of-life (EOL)</span>        | No support. The version can still be used but no bugfixes will be provided.                                                                                                  |
+
+#### Package Versioning
+
+Datadog APM for Ruby practices [semantic versioning](https://semver.org/).
+
+As this relates to downgrading runtime support, it implies:
+
+- **Major version updates** (e.g. `1.0.0` to `2.0.0`) may change support for any runtime
+  from [Beta](#support-beta)/[GA](#support-ga)
+  to [Maintenance](#support-maintenance)/[Legacy](#support-legacy)/[EOL](#support-eol).
+- **Minor version updates** (e.g. `1.0.0` to `1.1.0`) may change support for any runtime
+  from [Beta](#support-beta)/[GA](#support-ga) to [Maintenance](#support-maintenance).
+- **Patch version updates** (e.g. `1.0.0` to `1.0.1`) will not change support for any runtime.
+
+#### Library version support
+
+Datadog APM for Ruby will provide GA support for the latest major version and maintenance support for the previous major
+version of the library.
+
+| Gem Version | Support type                        |
+|-------------|-------------------------------------|
+| 2.x         | [GA](#support-ga)                   |
+| 1.x         | [Maintenance](#support-maintenance) |
+| 0.x         | [EOL](#support-eol)                 |
```

Comment:
For clarity, I feel there's a need to distinguish between "GA support" in the 2.x line. As is, it sounds like a hypothetical 2.0, 2.1, 2.2 would receive the exact same level of support as a hypothetical 2.3, i.e any bug fix that is landed on a 2.3.y are going to be all backported to 2.0.y, 2.1.y, 2.2.y.

Reply:
That's fair. What do you think is the best way to communicate that this only applies to the latest minor versions? I could change the column header `Gem Version` to `Gem Major Version` and drop the `.x`?

We could also add a sentence in the text above the table along the lines of `Respective levels of support for major versions are provided via the latest minor versions. Datadog will generally not backport changes to previous minor versions via patch releases. For example if 2.3 is the latest version of 2.x, changes will not be backported to 2.2, 2.1, etcetera. `

Reply:
I like the explicit nature of the expanded table, showing each version. You could collapse these into ranges where the info is repetitive e.g. `2.1-2.4`. But that's a minor issue.

Reply:
@delner As I understand it, this table is supposed to convey the level of support available in major versions. I agree with @lloeki that it's unclear in its current form that normally, that support is provided via new minor releases. Would listing every single version of the library help convey this information better?

Reply:
The annoying part with explicitly listing versions including minor is that we'd have to update the doc on every release, which would be annoying.

> `Gem Major Version`

That would have been my suggestion, but it also makes it less explicit that only the latest major is supported. 

Counterpoint though: that's what [node does](https://nodejs.org/en/about/previous-releases) and it seems fine.

I think a visual representation [such as this one](https://raw.githubusercontent.com/nodejs/Release/main/schedule.svg?sanitize=true) would reaaaally drive the point home.